Theodw 1530 create simple function to create horizon tables from schemas

KranthiRayipudi opened the pull request:
**PR Template**

 1. Are there new source to raw datasets?
	
	 - [ ] If yes - has a trigger been attached at the appropriate frequency?
  	 - [ ] No

 2. Have any new tables been created in Standardised?

  	- [ ] No
   	- [ ] If yes:
		- [ ] orchestration.json has been updated and tested in Dev and has been / is about to be PRd into main
		- [ ] the new schema exists inside *odw-config/standardised-table-definitions* OR is about to be PRd into main
			- Make sure to run Platform Integrate and Platform Deploy to Dev at least to ensure the schema is deployed into Synapse Dev Live 
		- [ ] Is the raw-to-standardised python script scheduled to run for this dataset grouping?
 3. Have any new tables been created in Harmonised or Curated?

   	 - [ ] No
     	 - [ ] If yes
	 	- [ ]  *2-odw-standardised-to-harmonised/py_odw_harmonised_table_creation* OR 4-*odw-harmonised-to-curated/py_odw_curated_table_creation* are set up to run in the pipeline *pln_post_deployments* with the base parameter specifying the correct table
		- [ ] the new schema exists inside *odw-config/harmonised-table-definitions* / *odw-config/curated-table-definitions* OR is about to be PRd into main
			- Make sure to run Platform Integrate and Platform Deploy to Dev at least to ensure the schema is deployed into Synapse Dev Live 
 4. Have any tables changed AND/OR have any columns changed in any scripts?
We only care about new columns or columns that change type.
	- [ ] No
 	- [ ] If yes:
		- [ ] Please set py_change_table to run in the pipeline *pln_post_deployments*
		- [ ] Please create a script to backdate and fill in this new column in Test and Prod
			Only delete and recreate tables with caution!
 4. Have any scripts run in isolation in dev? Please look at the *"spark.autotune.trackingId"*
		- If these changes need to be reflected in Test and Prod, please add to the pipeline *post-			deployment/pln_post_deployment*
	- [ ] Yes - I have reflected this script in the *pln_post_deployments* pipeline
	- [ ] Yes - This script is part of a scheduled run and has been added to the appropriate end to end pipeline with a trigger at the correct frequency
	- [ ] No - This change does not need to take place in Test and Prod
	- [ ] No - No scripts have run

RobPalmer30 reviewed (changes requested):
Please do the following:
- Ensure comments explain the flow of the code and what you are trying to achieve in each step
- How does the orchistration element ensure that only Horizon tables are picked up and not service bus tables by mistake?
- Organise code so that any parameters that need to be set in the post deployment step are defined in the first cell. Making it clear for all future developers. We should also use markdown in the cell above to explain what the parameters are
- Parameterise so that one notebook can be used for both standardised and harmonised for Horizon data
-We need to build out the json that is being referenced for the table structure, and not have URL as a parameter. Orchistration.json already has this defined in the standardised table definition parameter
- Extend config and notebook to also cover harmonised
- Review config and ensure we have table definitions for all Horizon tables
- Clean up code and remove code that is not used (e.g. imports not used, etc).

stef-solirius reviewed (changes requested):
Hi, great job! Just a thought:

Is it worth moving the get_latest_file() and generate_schema() funcitons inside:

py_0_source_to_raw_hr_functions instead of inside the notebook?

Rest looks great, happy to approve.

KranthiRayipudi commented:
•	Ensure comments explain the flow of the code and what you are trying to achieve in each step 
	I have commented each cell about code doing 
•	How does the orchistration element ensure that only Horizon tables are picked up and not service bus tables by mistake?
	I have added this step in code(the ‘Source_Folder' == "Horizon") for get the only Horizon table names from orchistration 
•	Organise code so that any parameters that need to be set in the post deployment step are defined in the first cell. Making it clear for all future developers. We should also use markdown in the cell above to explain what the parameters are
	 Need more clarification on this. please let me where I need to update  
•	Parameterise so that one notebook can be used for both standardised and harmonised for Horizon data
-We need to build out the Json that is being referenced for the table structure, and not have URL as a parameter. Orchestrations already has this defined in the standardised table definition parameter
	As per our initial discussion with Chris, we are currently merging the Service Bus and Horizon data into a single table in the harmonised layer. However, we've identified a need to maintain separate tables for Horizon data in the standardised layer. To address this, I've created a notebook to generate these tables in the standardised layer.
I would appreciate your guidance on a few points:
1.	Could you provide an example of a table that is currently using the harmonised layer?
2.	I've noticed that the schema in the standardised layer appears to be different from the harmonised layer. Could you confirm if this is expected?
3.	There seems to be a discrepancy between the current schema and the one defined in the config file. Could you advise on how we should proceed to ensure alignment?
•	Extend config and notebook to also cover harmonised
	You mean do I need identify the harmonised tables and create Json schema and create code for this.
•	Review config and ensure we have table definitions for all Horizon tables
	I have found few samples tables column names not matching with config and existing Horizon table (standardised and informed Chris and he created suppurate ticket for analysis 
•	Clean up code and remove code that is not used (e.g. imports not used, etc).
	I have removed

RobPalmer30 commented on workspace/notebook/create_Horizon_table_from_schema.json:
```diff
@@ -0,0 +1,495 @@
+				},
+				"source": [
+					"# Usage\r\n",
+					"url = f\"https://raw.githubusercontent.com/Planning-Inspectorate/odw-config/refs/heads/main/data-lake/standardised_table_definitions/Horizon/{entity_name}.json\"\r\n",
```
This was the parameter I was referring too that should be moved up. But it would be better for this to use the orchestration values rather than defined at this stage

KranthiRayipudi replied:
I have amended the code as suggested used standardised_Table_definition value stored in orchestration.json

RobPalmer30 commented:
I've added comments in line as appropriate.

Re Harmonised, I thought we had some appeals tables that still do. Let's not do the work to add the harmonised tables now - but instead ensure that it can be easily extended in future. Easiest way is to use the DB that's being called out with to work out the orchistration.json value we're looking for when pulling out the table definition. That way this approach can easily be extended to other areas in the future.
